server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -95,6 +101,48 @@ private NLPResult analyzeWithNLP(String text) throws IOException {
     return nlpResults;
   }
 
+  /**
+   * Takes the given map of sentiments and creates a common weighted 
+   * sentiment according to the number of likes each comment receives
+   * out of the sum of the likes of the comments
+   *
+   * @param unweightedNLPMap the unweighted map of <like, nlp> pairs
+   * @return A weighted NLPResult object
+   */
+  private NLPResult createWeightedSentiment(HashMap<NLPResult, Integer> unweightedNLPMap) {
+      double totalLikes = 0;
+
+      for (Map.Entry<NLPResult, Integer> likesAndNLPResult : unweightedNLPMap.entrySet()) {
+          totalLikes += (double) likesAndNLPResult.getValue();
+      }
+      
+      if (totalLikes == 0) {
+          // If there are no likes on any comment, we they all need
+          // to be weighted equally. There's not enough data to confidently
+          // determine a communities reaction. Setting totallikes++ to avoid error.
+          // This will rarely hit.
+          totalLikes++;
```

Comment:
return a zeroed  NLPResult instead of changing what totalLikes actually means. Setting totalLikes to 1 when the value it represents is actually 0 is confusing and makes your code brittle.